Src/Couchbase.Linq.IntegrationTests/QueryTests.cs

```diff
@@ -478,7 +478,7 @@ public async Task UseHashAndIndex_SelectDocuments()
             var context = new BucketContext(_travelSample);
 
             var query =
-                from route in context.Query<Route>()
+                from route in await CollectionSwitch<Route, RouteInCollection>(context)
```

Comment:
I agree it's odd, but it's really just a hack for the integration tests. That way they don't break when we run on pre-7.0 because it will query using the other POCO which doesn't have the CouchbaseCollection attribute. An alternative would be to duplicate the tests and apply a lot of skips to them, but it seemed a lot messier overall.

Perhaps if I just broke it out to a separate line, let me try that and see if it reads better.